src/main/resources/assets/carpet/scripts/draw.sc

```diff
@@ -0,0 +1,196 @@
+import('math','_euclidean_sq','_vec_length');
+
+__config() -> {
+    'commands'->{
+        'sphere <center> <radius> <block>'->['draw_sphere',null, false],
+        'sphere <center> <radius> <block> replace <replacement>'->['draw_sphere',false],
+        'ball <center> <radius> <block>'->['draw_sphere', null, true],
+        'ball <center> <radius> <block> replace <replacement>'->['draw_sphere', true],
+        'diamond <center> <radius> <block>'->['draw_diamond',null],
+        'diamond <center> <radius> <block> replace <replacement>'->'draw_diamond',
+        'cone <center> <radius> <height> <pointing> <orientation> <block> <hollow>'->['draw_pyramid',null, false],
+        'cone <center> <radius> <height> <pointing> <orientation> <block> <hollow> replace <replacement>'->['draw_pyramid', false],
+        'pyramid <center> <radius> <height> <pointing> <orientation> <block> <hollow>'->['draw_pyramid',null, true],
+        'pyramid <center> <radius> <height> <pointing> <orientation> <block> <hollow> replace <replacement>'->['draw_pyramid', true],
+        'cylinder <center> <radius> <height> <orientation> <block> <hollow>'->['draw_prism', null, false],
+        'cylinder <center> <radius> <height> <orientation> <block> <hollow> replace <replacement>'->['draw_prism', false],
+        'cuboid <center> <radius> <height> <orientation> <block> <hollow>'->['draw_prism', null, true],
+        'cuboid <center> <radius> <height> <orientation> <block> <hollow> replace <replacement>'->['draw_prism', true]
+    },
+    'arguments'->{
+        'center'->{'type'->'pos', 'loaded'->'true'},
+        'radius'->{'type'->'int', 'suggest'->[]},//to avoid default suggestions
+        'replacement'->{'type'->'block'},
+        'height'->{'type'->'int', 'suggest'->[]},
+        'orientation'->{'type'->'string', 'suggest'->['x','y','z']},
+        'pointing'->{'type'->'string','suggest'->['up','down']},
+        'hollow'->{'type'->'bool'}
+    },
+    'scope'->'global'
+};
+
+
+
+test(a,b,c)->print(center+ ' ' + radius + ' ' + block);
+
+//"Boilerplate" code
+
+set_block(x, y, z, block, replacement)-> (
+    success=0;
+    if(block != block(x, y, z),
+        set([x, y, z],block);
+        success+=1;
+
+        //todo add check for carpet setting CarpetSettings.fillUpdates
+        update(block(x, y, z))
+    );
+    success
+);
+
+length_sq(vec)->return(_vec_length(vec)^2);//cos of lengthSq func in DrawCommand.java
+
+fill_flat(pos, offset, dr, rectangle, orientation, block, hollow, replacement)->(
+    successes = 0;
+    r = floor(dr);
+    drsq = dr^2;
+    if(orientation=='x',
+        c_for(a=-r,a<=r,a+=1,
```

Comment:
these double forloops cannot be replaced by more efficient scans?

Reply:
u do realise that scan() function does the same thing right?

Reply:
It is literally a triple for loop, isn't it?

Reply:
Done

Reply:
ah wait thats fill_flat func, didnt do dat.

Reply:
ok did it now.